### ISSUE TITLE
Dark Mode

### ISSUE THREAD

**Issue by NickEichman** [OpenModelica/OpenModelica#9930]
Running OM version 1.20 in Fedora 35

Although it seems OMEdit directly uses dark/light mode according to system preference (dark mode in my case) through "Preserve user's GUI customizations", dark mode itself is hard to work with. Even going to tools -> options and adjusting the editor item colors, it is tough to use.
![Screenshot from 2022-12-12 14-16-59](https://user-images.githubusercontent.com/28605628/207054482-422ec84d-6cf7-414e-8cf6-c3bc11bf1561.png)

Additionally, the documentation browser is not adjustable through options.
![Screenshot from 2022-12-12 14-16-24](https://user-images.githubusercontent.com/28605628/207054376-1ad611c2-87ef-49ae-85ce-8d236e4b190c.png)

I was wondering if a dark mode palette is in the works/planned?

**Comment by casella** [comment]
@adeas31 I wonder if there is an easy way to make OMEdit generic with respect to these choices of GUI setup. What do you think?

**Comment by adeas31** [comment]
There is no easy way. We need to introduce concept of themes. And then we have to maintain color schemes of different themes.

**Comment by casella** [comment]
@NickEichman I'm afraid we currently have other far more pressing priorities, e.g. #8425 or #2891 😅 

If you are interested in contributing to the code base to add these features, please let us know.

**Comment by liulizhi1997** [comment]
@NickEichman  Hi, Can I reproduce your theme settings on Windows 11? I try but it doesn't work, the background color still white.......

**Comment by NickEichman** [comment]
@liulizhi1997 Unfortunately I never touched the Windows version of OMEdit, and have not touched OMEdit in general in a couple years, therefore am not aware what would be the best way of changing the settings. However, as you can see from my original post, even in v1.20 in Linux there were still issues.
Maybe if you find a way to alter it you could share for future people that might want to do the changes as well?

**Comment by liulizhi1997** [comment]
@NickEichman Thank you, I will try.

**Comment by casella** [comment]
@adeas31 I guess we need to schedule Dark Mode for the 1.27.0 release. 

Not entirely sure how diagrams will work with a white background, though. All the Modelica libraries were designed assuming a white canvas.

**Comment by adeas31** [comment]
> [@adeas31](https://github.com/adeas31) I guess we need to schedule Dark Mode for the 1.27.0 release.

Yeah. We can have dark mode once we switch completely to Qt6 as it provides better support for it.

> 
> Not entirely sure how diagrams will work with a white background, though. All the Modelica libraries were designed assuming a white canvas.

They will look a bit off. I tried the dark mode in Dymola and to be honest it doesn't look very nice to me. Seems they used a custom color scheme and not the one that is usually used with dark mode.

<img width="2282" height="1730" alt="Image" src="https://github.com/user-attachments/assets/755e63af-3a0e-4c05-9214-e2268c0cdcdb" />

**Comment by casella** [comment]
De gustibus non disputandum est 😃

**Comment by juantinazzo** [comment]
Any updates on this?

**Comment by adeas31** [comment]
> Any updates on this?

No. We are still working on the migration to Qt6 on Windows. The only remaining missing component is Qt WebEngine. I’ve spoken with @adrpo, and he will try to compile it. If that goes well, we can proceed with the switch to Qt6.

Once the Qt6 migration is complete, I will start looking into the remaining pieces needed for proper dark mode support. That said, it’s not possible to give an exact timeline. Progress depends on the number of other high-priority tasks.

**Comment by casella** [comment]
There will also be a tricky issue: the MSL icons (and also other library icons) were designed with a white background canvas in mind. I'm not sure how they'll look like in dark mode, probably not really well. In fact, Dymola has some kind of "grey mode" to cope with that.
